lab_1_keywords_tfidf/main.py

```diff
@@ -99,12 +161,32 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
 
     In case of corrupt input arguments, None is returned
     """
-    pass
-
+    # я понятия не имею работает ли эта функция
+    if isinstance(term_freq, dict) is False or isinstance(idf, dict) is False:
+        return None
+    else:
+        for k, v in term_freq.items():
+            if isinstance(k, str) is False or isinstance(v, float) is False:
+                return None
+        for k, v in term_freq.items():
+            if isinstance(k, str) is False or isinstance(v, float) is False:
+                return None
+        import math
+        tfidf_dict = {}
+        for key, v in term_freq.items():
+            if key in idf:
+                tfidf_dict[key] = v * idf.get(key)
+                """в словарь со значениями tfidf добавляю слово, которое является ключом, и его значение,
+                равное произведению значений данных на входе словарей(если вдруг я забуду что тут происходит )"""
+            else:
+                key_idf = math.log(47 / 0 + 1)
+                tfidf_dict[key] = v * key_idf
```

Comment:
it seems that you only add a new pair to the tfidf_dict when you get into `else` branch
you may want to reconsider identation here